Packs/Vectra/Integrations/VectraEventCollector/VectraEventCollector.py

```diff
@@ -0,0 +1,445 @@
+    _, detections = get_detections_cmd(client=client, first_timestamp=first_timestamp)
+    if detections:
+        next_run_detection = datetime.strptime(
+            detections[0].get("first_timestamp"), DETECTION_FIRST_TIMESTAMP_FORMAT  # type: ignore
+        )
+
+        # Need to add 1 minute since first_timestamp query parameter is inclusive
+        next_run_detection_str = (next_run_detection + timedelta(minutes=1)).strftime(
+            DETECTION_FIRST_TIMESTAMP_QUERY_START_FORMAT
+        )
+        demisto.info(f"{len(detections)} detections found.")
+    # If no detections were fetched, we can reuse the current first_timestamp
+    else:
+        next_run_detection_str = first_timestamp
+        demisto.info("No detections were found")
+
+    return (
+        detections,
+        audits,
+        {
+            DETECTION_NEXT_RUN_KEY: next_run_detection_str,
+            AUDIT_NEXT_RUN_KEY: next_run_audit_str,
+        },
+    )
+
+
+def get_events(
+    client: VectraClient, first_fetch: datetime
+) -> Tuple[CommandResults, List[Dict[str, Any]], CommandResults, List[Dict[str, Any]]]:
+
+    """
+    Command function to retrieve detections and audits.
+
+    Arguments:
+        - `client` (``VectraClient``): An instance of a Vectra API HTTP client.
+        - `first_fetch` (``datetime``): Parameter used as starting range to retrieve detections.
+
+    Returns:
+        - `CommandResults` of detections to War Room.
+        - `List[Dict[str, Any]]` of detections.
+        - `CommandResults` of audits to War Room.
+        - `List[Dict[str, Any]]` of audits.
+    """
+
+    detection_res, detections = get_detections_cmd(
+        client=client,
+        first_timestamp=first_fetch.strftime(DETECTION_FIRST_TIMESTAMP_QUERY_START_FORMAT),
+    )
+
+    audits_res, audits = get_audits_cmd(
+        client=client, start=first_fetch.strftime(AUDIT_START_TIMESTAMP_FORMAT)
+    )
+
+    return detection_res, detections, audits_res, audits
+
+
+""" MAIN FUNCTION """
+
+
+def main() -> None:  # pragma: no cover
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    cmd = demisto.command()
+    args = demisto.args()
+    config = demisto.params()
+
+    demisto.info(f"Command being called is '{cmd}'")
+    try:
+
+        client = VectraClient(
+            url=config.get("url"),
+            api_key=config.get("credentials", {}).get("password"),
+            fetch_limit=arg_to_number(arg=config.get("fetch_limit", 100)),  # type: ignore
+            insecure=config.get("insecure"),
+            proxy=config.get("proxy"),
+        )
+
+        if cmd == "test-module":
+            result = test_module(client)
+            return_results(result)
+
+        elif cmd in ("vectra-get-events", "fetch-events"):
+
+            if cmd == "vectra-get-events":
+                should_push_events = argToBoolean(args.pop("should_push_events"))
```

Comment:
No real reason, I was using the Event Collector template provided in https://github.com/demisto/content/pull/21981 and left it as is. Not sure it makes a difference since we don't have any other arguments to the `vectra-get-events` command.